lib/logr/entry.rb

```diff
@@ -0,0 +1,69 @@
+require "logr/event"
+require "logr/metric"
+
+module Logr
+  class Entry
+
+    attr_reader :metrics, :message
+
+    def initialize(logger, event=nil, metrics=[], message=nil)
+      @logger = logger
+      @event = event
+      @metrics = metrics
+      @message = message
+    end
+
+    def event(name, options={})
+      Entry.new(@logger, Event.new(name, options), @metrics)
+    end
+
+    def with(options={})
+      Entry.new(@logger, @event.with(options), @metrics)
+    end
+
+    def metric(name, value, type: "counter")
+      metric = Metric.new(name, value, type)
+      Entry.new(@logger, @event, @metrics + [metric])
+    end
+
+    def monitored(title, text)
+      event = @event.with(monitored: true, title: title, text: text)
```

Comment:
OK, I just figured out what's the purpose of `monitored`, it must come after an `event` call. This should be validated and a proper error raised, if it's not the case.